cmd/habitatctl/commands/community.go

```diff
@@ -44,23 +46,53 @@ var communityCreateCmd = &cobra.Command{
 	Short: "create a new community",
 	Run: func(cmd *cobra.Command, args []string) {
 
+		userIdentity, err := loadUserIdentity(cmd)
+		if err != nil {
+			printError(fmt.Errorf("error loading user identity: %s", err))
+		}
+
 		name := cmd.Flags().Lookup("name")
 		if name == nil {
-			fmt.Println("name flag needs to be set")
-			return
+			printError(errors.New("name flag needs to be set"))
 		}
 
 		ipfs, _ := cmd.Flags().GetBool("ipfs")
 
+		conn, err := getWebsocketConn(ctl.CommandCommunityCreate)
```

Comment:
why did we switch over to using websocket instead of just regular http? is it to preserve ordering of messages somehow? i feel like that could be better modeled with a state machine for generating cert / authing with server

Reply:
Yeah that would be another good way of doing it. I guess it was done this way mainly to preserve the existing API we have. The one advantage is like you said, its easier to keep track of the messages in this exchange over one http connection, rather than over multiple, where more weird race-condition-esque things could happen

Reply:
ah yeah that makes sense. i think its fine for now and we can revisit in the future. might be easier to do one http request at a time if everything is sequential though